pkg/manager/manager_tasks.go

```diff
@@ -236,12 +236,20 @@ func (s *singleton) Scan(input models.ScanMetadataInput) {
 
 			if err != nil {
 				logger.Errorf("Error encountered scanning files: %s", err.Error())
+				wg.Wait()                              // running tasks should finish and
+				instance.Paths.Generated.EmptyTmpDir() // tmp emptied
```

Comment:
This feels like the nuclear approach. I think its better to ensure that specific files are removed during the tasks, instead of clearing out the entire `tmp` directory.

Reply:
The scan task already does a `instance.Paths.Generated.EmptyTmpDir()` when its finished.
Doing the same when it exits due to an error  seemed natural to me.
I could just do a 
```golang
	if err != nil {
		logger.Errorf("Error encountered scanning files: %s", err.Error())
		break
		}
```
instead and add an extra if for the print in 
```golang
if s.Status.stopping
```

The curreent one singleton task at a time approach means that we should clear the tmp dir once that task is finished or not?

Reply:
The assumption that only one thread will be utilising `generated/tmp` can and probably will change in the future, so I'd rather be a little more surgical in how we remove files from it. As you say however, this is how it's done in other parts of the code, so I'm happy to leave it for now.

Reply:
Its not one thread but one singleton task. Threads or `smaller tasks` are more than one at a time. The weighted waitgroup that was added from the parallel tasks PR and is used in most of the tasks makes that possible. The generated/tmp is only cleaned after all the threads/small tasks are finished and then the scan is finished.